### ISSUE TITLE
BPA Analyzer crashes when creating tooltip

### ISSUE THREAD

**Issue by fmms** [otykier/TabularEditor#404]
Hi,

I have a model in which the BPA detects multiple issues. 
However, when I hover one of them it creates and exception:
![image](https://user-images.githubusercontent.com/908734/76700787-43c31f80-66bb-11ea-8ec4-bb7e94159f4f.png)

I am using version 2.9.5.

Thanks

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at TabularEditor.BestPracticeAnalyzer.AnalyzerResultTooltip.GetToolTip(TreeNodeAdv node, NodeControl nodeControl)
   at Aga.Controls.Tree.TreeViewAdv.GetNodeToolTip(TreeNodeAdvMouseEventArgs args)
   at Aga.Controls.Tree.TreeViewAdv.DisplayNodesTooltip(MouseEventArgs e)
   at Aga.Controls.Tree.TreeViewAdv.UpdateToolTip(MouseEventArgs e)
   at Aga.Controls.Tree.TreeViewAdv.OnMouseMove(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseMove(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3468.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
TabularEditor
    Assembly Version: 2.9.7376.23081
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3580.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3451.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
TOMWrapper
    Assembly Version: 2.9.7376.23078
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3580.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
Microsoft.AnalysisServices.Core
    Assembly Version: 18.4.0.5
    Win32 Version: 15.1.21.27
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/Microsoft.AnalysisServices.Core.DLL
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3580.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
Microsoft.AnalysisServices.Tabular
    Assembly Version: 18.4.0.5
    Win32 Version: 15.1.21.27
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/Microsoft.AnalysisServices.Tabular.DLL
----------------------------------------
Microsoft.AnalysisServices.Tabular.Json
    Assembly Version: 18.4.0.5
    Win32 Version: 15.1.21.27
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/Microsoft.AnalysisServices.Tabular.Json.DLL
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
Newtonsoft.Json
    Assembly Version: 10.0.0.0
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
System.Design
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Design/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Design.dll
----------------------------------------
FastColoredTextBox
    Assembly Version: 2.16.11.0
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
Crad.Windows.Forms.Actions
    Assembly Version: 1.3.0.0
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
Antlr4.Runtime
    Assembly Version: 4.5.0.0
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3324.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
Microsoft.WindowsAPICodePack
    Assembly Version: 1.1.2.0
    Win32 Version: 2.9.7376.23081
    CodeBase: file:///C:/Program%20Files%20(x86)/Tabular%20Editor/TabularEditor.exe
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3580.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.7.3468.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.



```

**Comment by otykier** [comment]
I'm not able to reproduce this. Could you share the Dynamic LINQ expression or the JSON definition in the BPARules.json file of the rule that causes the crash? Thanks!

**Comment by fmms** [comment]
Hi,
these rules have never been modified manually I suppose. Here it is:
![image](https://user-images.githubusercontent.com/908734/76861620-2928ac00-685d-11ea-8c08-30e12852abbb.png)

```
not (IsHidden or Table.IsHidden) 
and RegEx.IsMatch(Name, "[A-Z]([A-Z0-9]*[a-z][a-z0-9]*[A-Z]|[a-z0-9]*[A-Z][A-Z0-9]*[a-z])[A-Za-z0-9]*") 
and not Name.Contains(" ") 
and (Model.Cultures.Count = 0 or TranslatedNames.Any(it = "" or it = outerIt.Name))
```

**Comment by otykier** [comment]
Thanks. The crash is certainly a bug which will be fixed in the next version. In the mean time, please modify the expression as follows:

```
IsVisible 
and RegEx.IsMatch(Name, "[A-Z]([A-Z0-9]*[a-z][a-z0-9]*[A-Z]|[a-z0-9]*[A-Z][A-Z0-9]*[a-z])[A-Za-z0-9]*") 
and not Name.Contains(" ") 
and (Model.Cultures.Count = 0 or TranslatedNames.Any(it = "" or it = outerIt.Name))
```

**Comment by otykier** [comment]
Fixed in [2.9.6](https://github.com/otykier/TabularEditor/releases/tag/2.9.6).
